File: beibei-web/src/api/tag.ts
import { get, post, put, del } from '@/api/request'

export interface TagNode {
  id: number
  parentId: number
  name: string
  level: number
  sortOrder: number
  description: string
  origin: number
  chunkCount: number
  questionCount: number
  children: TagNode[]
}

export function getTagTree(kbId: number) {
  return get<TagNode[]>('/tag/tree', { kbId })
}

export function createTag(data: { kbId: number; parentId?: number; name: string; description?: string }) {
  return post<number>('/tag', data)
}

export function updateTag(id: number, data: { kbId: number; name?: string; description?: string; sortOrder?: number }) {
  return put<void>(`/tag/${id}`, data)
}

export function deleteTag(id: number) {
  return del<void>(`/tag/${id}`)
}

export function refreshTagCounts(kbId: number) {
  return post<void>('/tag/refresh', undefined, { params: { kbId } })
}

File: beibei-server/src/main/java/com/beibei/entity/Tag.java
package com.beibei.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 知识点树节点。表 bb_tag。最多 3 层，每个知识库独立一套。
 */
@Data
@TableName("bb_tag")
public class Tag implements Serializable {

    /** 来源：AI 抽取 */
    public static final int ORIGIN_AI = 1;
    /** 来源：手动创建 */
    public static final int ORIGIN_MANUAL = 2;

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long kbId;

    /** 0 表示根节点 */
    private Long parentId;

    private String name;

    /** 1 / 2 / 3 */
    private Integer level;

    private Integer sortOrder;

    private String description;

    private Integer origin;

    private Integer chunkCount;
    private Integer questionCount;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
